synapse/handlers/federation.py

```diff
@@ -3053,13 +3054,15 @@ async def persist_events_and_notify(
         """
         instance = self.config.worker.events_shard_config.get_instance(room_id)
         if instance != self._instance_name:
-            result = await self._send_events(
-                instance_name=instance,
-                store=self.store,
-                room_id=room_id,
-                event_and_contexts=event_and_contexts,
-                backfilled=backfilled,
-            )
+            # Limit the number of events sent over federation.
+            for batch in batch_iter(event_and_contexts, 50):
```

Comment:
```suggestion
            for batch in batch_iter(event_and_contexts, 1000):
```

We want to batch things up as match as possible, and I think 1000 should still be under the size limit

Reply:
Yeah I had no idea what to make this. I chose 50 as that's what comes in a transaction... wasn't sure what was reasonable though!